packages/block-library/src/query-pagination-numbers/index.php

```diff
@@ -91,14 +91,17 @@ function render_block_core_query_pagination_numbers( $attributes, $content, $blo
 	}
 
 	if ( $enhanced_pagination ) {
-		$p = new WP_HTML_Tag_Processor( $content );
+		$p         = new WP_HTML_Tag_Processor( $content );
+		$tag_index = 0;
 		while ( $p->next_tag(
-			array(
-				'tag_name'   => 'a',
-				'class_name' => 'page-numbers',
-			)
+			array( 'class_name' => 'page-numbers' )
 		) ) {
-			$p->set_attribute( 'data-wp-on--click', 'core/query::actions.navigate' );
+			if ( null === $p->get_attribute( 'data-wp-key' ) ) {
+				$p->set_attribute( 'data-wp-key', 'index-' . $tag_index++ );
```

Comment:
Why don't we use [the $page variable](https://github.com/WordPress/gutenberg/blob/f61a275613caea1b2d0f41a752272f4345814095/packages/block-library/src/query-pagination-numbers/index.php#L20) instead of creating our own tag_index?

Reply:
In this case, looking at the code:

```
$page_key = isset( $block->context['queryId'] ) ? 'query-' . $block->context['queryId'] . '-page' : 'query-page';
$page = empty( $_GET[ $page_key ] ) ? 1 : (int) $_GET[ $page_key ];
```

As far as I can remember, the queryId only appears when there is more than one query loop in the page. So, in most cases, `$page` will always return 1, having this HTML
![Screenshot 2024-02-12 at 16 22 46](https://github.com/WordPress/gutenberg/assets/37012961/7c8fe3d9-d11c-446e-8ba5-e5b481a31dbe)

With the same key, the fix won't work.